ReactAndroid/src/main/java/com/facebook/react/views/text/ReactFontManager.java

```diff
@@ -71,6 +83,15 @@ public static ReactFontManager getInstance() {
     return typeface;
   }
 
+  public void loadFont(Context context, int fontId) {
+    Typeface font = ResourcesCompat.getFont(context, fontId);
+    if (font != null) {
+      String fontFamily = context.getResources().getResourceEntryName(fontId);
```

Comment:
This won't affect FB with this PR, but I wonder if this is efficient in practice. You don't have to address it in this PR, but I think allowing apps to provide a custom "typeface resolution" function can allow them to address slow lookup. For instance, if one day FB wants to use custom fonts, I'd actually implement the resolver via a switch statement, not via asking `getResourceEntryName()`. Most apps probably don't need such optimization, but having an option to do optimized thing is good.

Reply:
> Most apps probably don't need such optimization, but having an option to do optimized thing is good.

An example pseudo solution:

```
public interface ReactTypefaceResolver {
  public String resolveTypefaceName(int fontId);
}
```

Then introduce a method in ReactFontManager:

```
public void setCustomTypefaceResolver(ReactTypefaceResolver resolver) {
  mCustomTypefaceResolver = resolver;
}
```

Then you can just use `mCustomTypefaceResolver != null` as the flag, no need to have the boolean flag you had above. Then you can just call it inside `loadFont()`.